src/components/App/App.js

```diff
@@ -1,62 +1,187 @@
 import Footer from '../Footer/Footer';
 import Login from '../Login/Login';
 import InfoTooltip from '../InfoTooltip/InfoTooltip';
 import Register from '../Register/Register';
+import { CurrentUserContext } from '../../contexts/CurrentUserContext';
+import ProtectedRoute from '../ProtectedRoute/ProtectedRoute';
 import './App.css';
 
 const App = () => {
 
   const [loggedIn, setLoggedIn] = React.useState(false);
+  const [currentUser, setCurrentUser] = React.useState({});
+  const [name, setName] = React.useState('');
+
   const [isLoginPopupOpen, setIsLoginPopupOpen] = React.useState(false);
   const [isRegisterPopupOpen, setIsRegisterPopupOpen] = React.useState(false);
   const [isInfoTooltipOpen, setIsInfoTooltipOpen] = React.useState(false);
-  // const [cardListOpen, setCardListOpen] = React.useEffect(false);
-
-  const [searchSubmit, setSearchSubmit] = React.useState(false);
  
-  // const [textField, setTextField] = React.useState('');
+  // Articles states
+  const [isDataLoading, setIsDataLoading] = React.useState(false);
+  const [searchSubmit, setSearchSubmit] = React.useState(false);
+  
+  const [keyword, setKeyword] = React.useState(localStorage.getItem('keyword'));
+  const [articles, setArticles] = React.useState(JSON.parse(localStorage.getItem('searchResults')))
+  const [savedArticles, setSavedArticles] = React.useState([]);
 
-  const handleSearchSubmit = () => {
-    setSearchSubmit(true);
+  
+  const navigate = useNavigate();
 
-  }
+  React.useEffect(() => {
+    loggedIn && 
+      mainApi
+        .getUserData()
+        .then((res) => setCurrentUser(res))
+        .catch((err) => console.log(`Error.....: ${err}`));
+  }, [loggedIn]);
 
-  // const navigate = useNavigate();
+const getArticles = useCallback(() => {
+  loggedIn && 
+    mainApi
+      .getSavedArticles()
+      .then((articlesList) => {
+        setSavedArticles(
+          articlesList.map((article) => {
+            return {
+              _id: article._id,
+              keyword: article.keyword,
+              title: article.title,
+              content: article.text,
+              publishedAt: article.date,
+              source: {name: article.source},
+              url: article.link,
+              urlToImage: article.image,
+              owner: article.owner,
+            }
+          })
+        );
+      })
+        .catch(() => setSavedArticles([]));
+      }, [loggedIn]);
   
-  const handleUserLogin = (email, password) => {
-    if (!email || !password) {
-      return console.log('Error.....');
-    }
-    setLoggedIn(true);
+  React.useEffect(() => {
+    getArticles();
+  } ,[getArticles]);
 
-    closeAllPopups();
+  // save article of the current user
+  const handleSaveArticle = (article) => {
+    const isSavedAlready = savedArticles.find((savedArticle) => savedArticle.url === article.url);
+    if (isSavedAlready) {
+      mainApi
+        .deleteArticle(isSavedAlready._id)
+        .then(() => getArticles())
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**